File: app/src/main/java/com/live/myapplication/MainActivity.java
package com.live.myapplication;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.text.SimpleDateFormat;
import java.util.Date;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    private LiveDataViewModel viewModel;
    EditText editInterval;
    Button btnStart;
    Boolean m_bStartFlag = true;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editInterval = (EditText) findViewById(R.id.edit_interval);
        btnStart = (Button) findViewById(R.id.btn_start);
        btnStart.setOnClickListener(this);

    }
    private void subscribe() {
        final Observer<Long> delayTimeObserver = new Observer<Long>() {
            @Override
            public void onChanged(@Nullable final Long aLong) {
                SimpleDateFormat sdf = new SimpleDateFormat("MMM dd,yyyy HH:mm:ss");
                Date resultdate = new Date(aLong);
                String newText = sdf.format(resultdate);
                ((TextView) findViewById(R.id.tm_result)).setText(newText);
            }
        };

        viewModel.getDelayTime().observe(this, delayTimeObserver);
    }

    @Override
    public void onClick(View view) {
        if(m_bStartFlag)
        {
            int ninterval = 0;
            try {
                ninterval = Integer.parseInt(String.valueOf(editInterval.getText()));
            }catch (Exception er){
                return;
            }

            m_bStartFlag = false;
            editInterval.setEnabled(false);
            btnStart.setText("Stop");

            viewModel = new LiveDataViewModel(ninterval);
            subscribe();
        }
        else
        {
            btnStart.setText("Start");
            m_bStartFlag = true;
            editInterval.setEnabled(true);

            viewModel.onCleared();
        }
    }
}

File: app/src/main/java/com/live/myapplication/MainActivity.kt
package com.live.myapplication

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import java.text.SimpleDateFormat
import java.util.*

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private var viewModel: LiveDataViewModel? = null
    var editInterval: EditText? = null
    var btnStart: Button? = null
    var m_bStartFlag = true
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        editInterval = findViewById<View>(R.id.edit_interval) as EditText
        btnStart = findViewById<View>(R.id.btn_start) as Button
        btnStart!!.setOnClickListener(this)
    }

    private fun subscribe() {
        val delayTimeObserver: Observer<Long?> = Observer<Long?> { aLong ->
            val sdf = SimpleDateFormat("MMM dd,yyyy HH:mm:ss")
            val resultdate = Date(aLong!!)
            val newText = sdf.format(resultdate)
            (findViewById<View>(R.id.tm_result) as TextView).text = newText
        }
        viewModel!!.delayTime.observe(this, delayTimeObserver)
    }

    override fun onClick(view: View) {
        if (m_bStartFlag) {
            var ninterval = 0
            ninterval = try {
                editInterval!!.text.toString().toInt()
            } catch (er: Exception) {
                return
            }
            m_bStartFlag = false
            editInterval!!.isEnabled = false
            btnStart!!.text = "Stop"
            viewModel = LiveDataViewModel(ninterval.toLong())
            subscribe()
        } else {
            btnStart!!.text = "Start"
            m_bStartFlag = true
            editInterval!!.isEnabled = true
            viewModel!!.onCleared()
        }
    }
}